src/axom/inlet/docs/sphinx/advanced_types.rst

```diff
@@ -1,55 +1,152 @@
-for convenience.  The schema definition for a class or struct could also be implemented as a
-free function for third-party types, or even in the same place as the sub-table declaration. 
+.. note:: 
+  The definition of a static ``defineSchema`` member function is not required, and is just used
+  for convenience.  The schema definition for a class or struct could also be implemented as a
+  free function for third-party types, or even in the same place as the sub-table declaration.
+
+Accessing
+---------
+
+In order to convert from Inlet's internal representation to a custom C++ ``struct``, you must provide 
+deserialization logic.  This is accomplished by a specialization of the ``FromInlet<T>`` functor for your
+type ``T``, which implements a member function with the signature ``T operator()(const inlet::Table&)``.
+This function should return an instance of type ``T`` with its members populated with the corresponding
+fields in the input file. For the simple ``Car`` example whose schema is defined above,the specialization
+might look like:
+
+.. literalinclude:: ../../examples/user_defined_type.cpp
+   :start-after: _inlet_userdef_simple_frominlet_start
+   :end-before: _inlet_userdef_simple_frominlet_end
+   :language: C++
+
+In the above snippet, ``Table::operator[]`` is used to extract data from Inlet's internal representation
+which is automatically converted to the correct member variable types when the function's return value is
+constructed.  This conversion does not happen automatically for user-defined types.
+If a ``Car`` object as defined above is located at the path "car" within the input file, it can be retrieved as follows:
+
+.. code-block:: C++
+
+  Car car = inlet["car"].get<Car>();
+
+**********************************
+Arrays and Dictionaries of Structs
+**********************************
+
+Arrays of user-defined types are also supported in Inlet.  
+
+Defining And Storing
+--------------------
 
-Arrays of user-defined types are also supported in Inlet.  First, use the ``addStructArray``
-function to create a subtable, then define the schema on that table:
+Consider a collection of cars, described in Lua as:
+
+.. code-block:: Lua
+
+  fleet = {
+    {
+      make = "Globex Corp",
+      seats = 3,
+      horsepower = 155,
+      color = "green"
+    },
+    {
+      make = "Initech",
+      seats = 4,
+      horsepower = 370
+      -- uses default value for color
+    },
+    {
+      make = "Cyberdyne",
+      seats = 1,
+      horsepower = 101,
+      color = "silver"
+    }
+  }
+
+or in YAML, as:
+
+.. code-block:: YAML
+
+  fleet:
+    - make: Globex Corp
+      seats: 3
+      horsepower: 155
+      color: green
+    - make: Initech
+      seats: 4
+      horsepower: 370
+      # uses default value for color
+    - make: Cyberdyne
+      seats: 1
+      horsepower: 101
+      color: silver
+
+First, use the ``addStructArray`` function to create a subtable, then define the schema on that table:
```

Comment:
It might be worth mentioning/noting that this uses the same ``Car::addSchema()`` function as used for a single car (above).